components/brave_wallet/browser/polkadot/polkadot_extrinsic.rs

```diff
@@ -0,0 +1,202 @@
+            mut input: &[u8],
+            pubkey: &mut [u8; 32],
+            send_amount: &mut [u8; 16],
+        ) -> bool;
+    }
+}
+
+struct ChainMetadata {
+    metadata: HashMap<String, MetaData>,
+}
+
+struct MetaData {
+    balances_pallet_index: u8,
+    transfer_allow_death_call_index: u8,
+}
+
+fn make_chain_metadata() -> Box<ChainMetadata> {
+    Box::new(ChainMetadata { metadata: HashMap::default() })
+}
+
+impl ChainMetadata {
+    fn add_chain_metadata(self: &mut ChainMetadata, chain_id: &str, chain_name: &str) -> bool {
+        if self.metadata.contains_key(chain_id) {
+            return true;
+        }
+
+        let meta_data = match chain_name {
+            "Westend" => MetaData {
+                balances_pallet_index: POLKADOT_TESTNET_BALANCES_PALLET,
+                transfer_allow_death_call_index: POLKADOT_TESTNET_TRANSFER_ALLOW_DEATH_CALL_INDEX,
+            },
+            "Westend Asset Hub" => MetaData {
+                balances_pallet_index: POLKADOT_ASSET_HUB_TESTNET_BALANCES_PALLET,
+                transfer_allow_death_call_index:
+                    POLKADOT_ASSET_HUB_TESTNET_TRANSFER_ALLOW_DEATH_CALL_INDEX,
+            },
+            "Polkadot Asset Hub" => MetaData {
+                balances_pallet_index: POLKADOT_ASSET_HUB_MAINNET_BALANCES_PALLET,
+                transfer_allow_death_call_index:
+                    POLKADOT_ASSET_HUB_MAINNET_TRANSFER_ALLOW_DEATH_CALL_INDEX,
+            },
+            "Polkadot" => MetaData {
+                balances_pallet_index: POLKADOT_MAINNET_BALANCES_PALLET,
+                transfer_allow_death_call_index: POLKADOT_MAINNET_TRANSFER_ALLOW_DEATH_CALL_INDEX,
+            },
+            _ => return false,
+        };
+
+        self.metadata.insert(chain_id.to_string(), meta_data);
+        true
+    }
+
+    fn has_chain_metadata(self: &ChainMetadata, chain_id: &str) -> bool {
+        self.metadata.contains_key(chain_id)
+    }
+}
+
+fn encode_unsigned_transfer_allow_death(
+    chain_data: &ChainMetadata,
+    chain_id: &str,
+    send_amount_bytes: &[u8; 16],
+    pubkey: &[u8; 32],
+) -> Vec<u8> {
+    let Some(MetaData { balances_pallet_index, transfer_allow_death_call_index }) =
+        chain_data.metadata.get(chain_id)
+    else {
+        return Vec::new();
+    };
+
+    let mut buf = vec![
+        EXTRINSIC_VERSION,
+        *balances_pallet_index,
+        *transfer_allow_death_call_index,
+        MULTIADDRESS_TYPE,
+    ];
+
+    buf.extend_from_slice(pubkey);
+
+    Compact(u128::from_le_bytes(*send_amount_bytes)).encode_to(&mut buf);
+
+    Compact(buf.len() as u64).using_encoded(|encoded_len| {
+        buf.splice(0..0, encoded_len.iter().copied());
+    });
+
+    buf
+}
+
+fn decode_unsigned_transfer_allow_death(
+    chain_data: &ChainMetadata,
+    chain_id: &str,
+    mut input: &[u8],
+    pubkey: &mut [u8; 32],
+    send_amount_bytes: &mut [u8; 16],
+) -> bool {
+    let Ok(len) = Compact::<u64>::decode(&mut input) else {
```

Comment:
Why not to follow same Result approach as in sr25519?

Reply:
Interestingly, the sr25519.rs that I wrote doesn't seem to use a Result pattern. But I've removed the possibility of failure by removing the `chain_id` lookup so now we know for certain: if we hit a `false` it's because there's invalid octets somewhere.